docs/cookbook/server-managed-meta-tags.md

```diff
@@ -0,0 +1,351 @@
+
+  onMount(() => {
+    // Clean up server-rendered tags
+    document.head
+      .querySelectorAll('[inertia]')
+      .forEach(el => el.remove());
+
+    ready = true;
+  });
+</script>
+
+<svelte:head>
+  {#if ready}
+    <!-- Void elements (no content) -->
+    {#each voidTags as tag (tag.headKey)}
+      <svelte:element
+        this={tag.tagName}
+        inertia={tag.headKey}
+        {...tag.attrs}
+      />
+    {/each}
+
+    <!-- Elements with content -->
+    {#each contentTags as tag (tag.headKey)}
+      <svelte:element
+        this={tag.tagName}
+        inertia={tag.headKey}
+        {...tag.attrs}
+      >
+        {@html
+          typeof tag.innerContent === 'string'
+            ? tag.innerContent
+            : JSON.stringify(tag.innerContent)
+        }
+      </svelte:element>
+    {/each}
+  {/if}
+</svelte:head>
+
+```
+:::
+
+## Rendering Meta Tags
+
+Tags are defined as plain hashes and conform to the following structure:
+
+```ruby
+# All fields are optional.
+{
+  # Defaults to "meta" if not provided
+  tag_name: "meta",
+
+  # Used for <meta http-equiv="...">
+  http_equiv: "Content-Security-Policy",
+
+  # Used to deduplicate tags. InertiaRails will auto-generate one if not provided
+  head_key: "csp-header",
+
+  # Used with <script>, <title>, etc.
+  inner_content: "Some content",
+
+  # Any additional attributes will be passed directly to the tag.
+  # For example: name: "description", content: "Page description"
+  name: "description",
+  content: "A description of the page"
+}
+```
+The `<title>` tag has shortcut syntax:
+
+```ruby
+{ title: "The page title" }
+```
+
+### In the renderer
+
+Add meta tags to an action by passing an array of hashes to the `meta:` option in the `render` method:
+
+```ruby
+class EventsController < ApplicationController
+  def show
+    event = Event.find(params[:id])
+
+    render inertia: 'Event/Show', props: { event: event.as_json }, meta: [
+      { title: "Check out the #{event.name} event!" },
+      { name: 'description', content: event.description },
+      { tag_name: 'script', type: 'application/ld+json', inner_content: { '@context': 'https://schema.org', '@type': 'Event', name: 'My Event' } }
+    ]
+  end
+end
+```
+
+### Shared Meta Tags
+
+Often, you will want to define default meta tags that are shared across certain pages and which you can override within a specific controller or action. Inertia Rails has an `inertia_meta` controller instance method which references a store of meta tag data.
+
+You can call it anywhere in a controller to manage common meta tags, such as in `before_action` callbacks or directly in an action.
+
+```ruby
+class EventsController < ApplicationController
+  before_action :set_meta_tags
+
+  def show
+    render inertia: 'Event/Show', props: { event: Event.find(params[:id]) }
+  end
+
+  private
+
+  def set_meta_tags
+    inertia_meta.add([
+      { title: 'Look at this event!' }
+    ])
+  end
+end
+```
+
+#### The `inertia_meta` API
```

Comment:
This is basically the API I _wish_ we had built for `inertia_share`. If you look at the commit history, I actually built out an entire implementation with `inertia_meta` as a controller class method, before remembering how much I regret doing that for `inertia_share` 😅